examples/cis3-nft-sponsored-txs/tests/tests.rs

```diff
@@ -304,19 +780,77 @@ fn initialize_contract_with_alice_tokens() -> (Chain, ContractAddress, ContractI
         })
         .expect("Mint tokens");
 
-    (chain, contract_address, update)
+    (chain, contract_address, update, keypairs)
 }
 
 /// Setup chain and contract.
 ///
-/// Also creates the two accounts, Alice and Bob.
+/// Also creates the three accounts, Alice, Bob, and Charlie.
 ///
 /// Alice is the owner of the contract.
-fn initialize_chain_and_contract() -> (Chain, ContractAddress) {
+/// Alice's account is created with keys.
+/// Hence, Alice's account signature can be checked in the test cases.
+fn initialize_chain_and_contract(
+    generate_keys: bool,
+) -> (Chain, ContractAddress, Option<AccountKeys>) {
     let mut chain = Chain::new();
 
+    let (account_access_structure, keypairs) = match generate_keys {
+        // If `generate_keys` is true, fresh keys are generated for Alice.
+        // Since Alice's private key is available, Alice can sign and generate a valid signature in
+        // the test cases.
+        true => {
+            let rng = &mut rand::thread_rng();
+
+            let keypairs = AccountKeys::singleton(rng);
+            ((&keypairs).into(), Some(keypairs))
+        }
+        // If `generate_keys` is false, Alice's account is assigned a hardcoded public key.
+        // Since Alice's private key is NOT available, hardcoded signatures are used in the test
+        // cases. The signatures are generated outside the test cases (e.g. with https://cyphr.me/ed25519_tool/ed.html).
+        false => {
+            let mut inner_key_map: BTreeMap<KeyIndex, VerifyKey> = BTreeMap::new();
+
+            inner_key_map.insert(
+                KeyIndex(0u8),
+                VerifyKey::Ed25519VerifyKey(
+                    ed25519_dalek::PublicKey::from_bytes(&PUBLIC_KEY)
+                        .expect("Should be able to create public key"),
+                ),
+            );
+
+            let credential_public_keys = CredentialPublicKeys {
+                keys:      inner_key_map,
+                threshold: SignatureThreshold::ONE,
+            };
+
+            let mut key_map: BTreeMap<CredentialIndex, CredentialPublicKeys> = BTreeMap::new();
+            key_map.insert(
+                CredentialIndex {
+                    index: 0u8,
+                },
+                credential_public_keys,
+            );
```

Comment:
Like we have the `singleton` method on AccountKeys, we should have either a singleton method on AccountAccessStructure that constructs it from a public key, or add a From instance.

Reply:
Added `singleton` method.